src/user_crate/mod.rs

```diff
@@ -71,24 +71,27 @@ impl UserCrate<StateProvisioned> {
             db_oid = %self.0.db_oid(),
             fn_oid = %self.0.fn_oid(),
             crate_dir = %self.0.crate_dir().display(),
-            target_dir = target_dir.map(|v| tracing::field::display(v.display())),
+            target_dir = tracing::field::display(target_dir.display()),
         ))]
     pub fn build(
         self,
-        artifact_dir: &Path,
         pg_config: PathBuf,
-        target_dir: Option<&Path>,
+        target_dir: &Path,
     ) -> eyre::Result<(UserCrate<StateBuilt>, Output)> {
         self.0
-            .build(artifact_dir, pg_config, target_dir)
+            .build(pg_config, target_dir)
             .map(|(state, output)| (UserCrate(state), output))
     }
+
+    pub(crate) fn crate_dir(&self) -> &Path {
+        self.0.crate_dir()
+    }
 }
 
 impl UserCrate<StateBuilt> {
     #[tracing::instrument(level = "debug")]
-    pub(crate) fn built(db_oid: pg_sys::Oid, fn_oid: pg_sys::Oid, shared_object: PathBuf) -> Self {
-        UserCrate(StateBuilt::new(db_oid, fn_oid, shared_object))
+    pub(crate) fn built(db_oid: pg_sys::Oid, fn_oid: pg_sys::Oid, shared_object: &Path) -> Self {
+        UserCrate(StateBuilt::new(db_oid, fn_oid, shared_object.to_path_buf()))
```

Comment:
If we know we're just going to allocate anyways, then this change just prompts a reallocation, and it's reasonable to ask the caller to use `.to_owned()` on their `&Path`. This should remain as it was.

```suggestion
    pub(crate) fn built(db_oid: pg_sys::Oid, fn_oid: pg_sys::Oid, shared_object: PathBuf) -> Self {
        UserCrate(StateBuilt::new(db_oid, fn_oid, shared_object))
```